controllers/argocd/secret.go

```diff
@@ -184,7 +184,7 @@ func (r *ReconcileArgoCD) reconcileArgoSecret(cr *argoprojv1a1.ArgoCD) error {
 		common.ArgoCDKeyTLSPrivateKey:      tlsSecret.Data[common.ArgoCDKeyTLSPrivateKey],
 	}
 
-	if cr.Spec.Dex != nil || (cr.Spec.SSO != nil && cr.Spec.SSO.Provider == v1alpha1.SSOProviderTypeDex) {
+	if cr.Spec.SSO != nil && cr.Spec.SSO.Provider.ToLower() == v1alpha1.SSOProviderTypeDex.ToLower() {
```

Comment:
@svghadi the case conversion is only required on `spec.sso.provider` since that is coming from the user. `v1alpha1.SSOPRoivderTypeDex` is already defined in lower case so this is not required
Let's try to simplify where we can

Reply:
Here, `.ToLower()` on `v1alpha1.SSOPRoivderTypeDex` is used to resolve type mismatch error
> invalid operation: cr.Spec.SSO.Provider.ToLower() == v1alpha1.SSOProviderTypeDex (mismatched types string and "github.com/argoproj-labs/argocd-operator/api/v1alpha1".SSOProviderType)

But I agree, we should remove unwanted calls. I will try to simplify by updating `ToLower()` function to return `SSOProviderType` instead of `string` so that we can remove extra call on already defined values.